### ISSUE TITLE
flaky test failure in REST

### ISSUE THREAD

**Issue by gupadhyaya** [rollkit/rollkit#1798]
```
=== NAME  TestREST
    service_test.go:99: 
        	Error Trace:	/home/runner/work/rollkit/rollkit/rpc/json/service_test.go:99
        	Error:      	"{\"jsonrpc\":\"2.0\",\"result\":{\"block_id\":{\"hash\":\"CA7F7790577FC09E303816CAC94201914615260D3097D1A04B4BDA0F0377AA45\",\"parts\":{\"total\":0,\"hash\":\"\"}},\"block\":{\"header\":{\"version\":{\"block\":\"11\"},\"chain_id\":\"test\",\"height\":\"1\",\"time\":\"2024-08-14T19:21:25.972545451Z\",\"last_block_id\":{\"hash\":\"\",\"parts\":{\"total\":0,\"hash\":\"\"}},\"last_commit_hash\":\"EA312FCB12D4AEB315427DE61F738D6790DF83E49A83108F1B745947598950B4\",\"data_hash\":\"6E340B9CFFB37A989CA544E6BB780A2C78901D3FB33738768511A30617AFA01D\",\"validators_hash\":\"0554D6F24D8529FF529B4337E519C29516249441C316719E99E1A2D088A4A02B\",\"next_validators_hash\":\"0554D6F24D8529FF529B4337E519C29516249441C316719E99E1A2D088A4A02B\",\"consensus_hash\":\"0000000000000000000000000000000000000000000000000000000000000000\",\"app_hash\":\"\",\"last_results_hash\":\"E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855\",\"evidence_hash\":\"E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855\",\"proposer_address\":\"5E77C57BE649671B2DE2B019A910E8FFF1CFC5DF\"},\"data\":{\"txs\":[]},\"evidence\":{\"evidence\":null},\"last_commit\":{\"height\":\"1\",\"round\":0,\"block_id\":{\"hash\":\"CA7F7790577FC09E303816CAC94201914615260D3097D1A04B4BDA0F0377AA45\",\"parts\":{\"total\":0,\"hash\":\"\"}},\"signatures\":[{\"block_id_flag\":2,\"validator_address\":\"5E77C57BE649671B2DE2B019A910E8FFF1CFC5DF\",\"timestamp\":\"2024-08-14T19:21:25.972545451Z\",\"signature\":\"lVbbYWu4WLMo6t0+HaQfobd/Fy6NfjQ+63jlysqxqIunaNHx4jZCrj7rcxguG4a5GUouJefw/ILKpzSlmuvzDw==\"}]}}},\"id\":-1}\n" does not contain "failed to load hash from index"
        	Test:       	TestREST
    service_test.go:103: 
        	Error Trace:	/home/runner/work/rollkit/rollkit/rpc/json/service_test.go:103
        	Error:      	Expected value not to be nil.
        	Test:       	TestREST
=== NAME  TestREST/valid/missing_optional_param
    testing.go:1576: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
--- FAIL: TestREST (0.06s)
```

**Comment by tzdybal** [comment]
Tried to reproduce with: "go test ./rpc/json/ -run TestREST -count 1000" - without luck :thinking:

**Comment by tzdybal** [comment]
@gupadhyaya do you remember what version of rollkit was affected? Are you able to reproduce the issue?

TBH the only thing that comes to my mind is ["loopvar"](https://go.dev/blog/loopvar-preview), but this was fixed with go 1.22 (and we've [adopted it in May](https://github.com/rollkit/rollkit/commit/cceb2d2eb9f667ca25932ece006c5b19a18de746)).

**Comment by tuxcanfly** [comment]
@tzdybal added steps to repro here: https://github.com/rollkit/rollkit/pull/1880#issuecomment-2400061117
